lib/apiservers/engine/backends/system.go

```diff
@@ -159,31 +159,31 @@ func (s *System) SystemInfo() (*types.Info, error) {
 	if err != nil || vchInfo == nil {
 		log.Infof("System.SystemInfo unable to get vch info from port layer: %s", err.Error())
 	} else {
-		if vchInfo.CPUMhz != nil {
-			info.NCPU = int(*vchInfo.CPUMhz)
+		if vchInfo.CPUMhz > 0 {
+			info.NCPU = int(vchInfo.CPUMhz)
 
 			customInfo := [2]string{systemStatusMhz, fmt.Sprintf("%d Mhz", info.NCPU)}
 			info.SystemStatus = append(info.SystemStatus, customInfo)
 		}
-		if vchInfo.Memory != nil {
-			info.MemTotal = *vchInfo.Memory << 20 //Multiply by 1024*1024 to get Mebibytes
+		if vchInfo.Memory > 0 {
+			info.MemTotal = vchInfo.Memory * 1024 * 1024 // Get Mebibytes
```

Comment:
ok, if go compiler is smart to do this.  Still hurts the eyes of former C/C++ programmers. :)